Hephaestus.Common/Utilities/ConsoleUtility.cs

```diff
@@ -4,115 +4,96 @@
 
-                if (key == "n")
+                if (response != "y" || response != "yes" || response != "n" || response != "no")
                 {
-                    result = false;
-                    break;
+                    Console.WriteLine($"{response} is not a valid anwser to the given question: {question}. Please enter either Y for yes or N for no.");
+                    continue;
                 }
-
-                Console.WriteLine("Incorrect selection. Try again.");
+                break;
             }
-
-            return result;
+            return response == "y" || response == "yes" ? true : false;
         }
 
         /// <summary>
-        /// Ask the user to enter a string.
+        /// Prints a given message to the console and then prompts the user to enter a string.
         /// </summary>
-        /// <param name="message">Message displayed to the user. Message is transformed into 'messageHere: '.</param>
-        /// <returns></returns>
+        /// <param name="message">String to output to the console and act as a message to the user. The provided string is outputed in the following format: 'messageHere: '.</param>
+        /// <returns>Returns the entered string</returns>
         public static string AskToEnterString(string message)
         {
             Console.Write($"{message}: ");
-
-            string enteredString = Console.ReadLine();
-
-            return enteredString;
+            return Console.ReadLine();
         }
 
         /// <summary>
-        /// Ask the user to enter either a directory or a file path.
+        /// Prints a given message to the console and then prompts the user to enter a filesystem path corresponding to either a directory or a file.
         /// </summary>
-        /// <param name="message">Message displayed to the user. Message is transformed into 'messageHere: '.</param>
-        /// <param name="pathType">Select what type of path should the user enter.</param>
+        /// <param name="message">String to output to the console and act as a message to the user. The provided string is outputed in the following format: 'messageHere: '.</param>
+        /// <param name="pathType">The type of path the user needs to enter. Either a directory or an individual file.</param>
         /// <returns>Path to the directory or file.</returns>
-        public static string AskToEnterPath(string message, PathType pathType)
+        public static string AskToEnterFileSystemPath(string message, PathType pathType)
         {
-            string enteredString;
+            string enteredPath;
+            Console.Write($"{message}: ");
+
             while (true)
             {
-                Console.Write($"{message}: ");
-                
-                enteredString = Console.ReadLine();
-
-                if (pathType == PathType.Directory)
+                enteredPath = Console.ReadLine();
+                if (Directory.Exists(enteredPath) || File.Exists(enteredPath))
```

Comment:
I'd prefer explicit checking for Directory **or** File. As some inputs require only directory, not both which could lead to errors further down. A user might confuse private key with the directory where it's stored not that it actually requires a full path including the name. As it's not checked here, it will throw an exception later down.

Reply:
Explicit as in branching both checks into separate conditional statements? Like how it was prior to this pull request?

Reply:
So, let me put it this way.
Hephaestus is asking a user to enter a **file path** to the private key but the user enters a directory. By this validation it would pass even if it’s asked only to input file path.

Reply:
Corrected my mistake